WcaOnRails/app/models/competition_event.rb

```diff
@@ -21,4 +21,31 @@ def to_wcif
       "rounds" => self.rounds.map(&:to_wcif),
     }
   end
+
+  def load_wcif!(wcif)
+    # Remove extra rounds.
+    self.rounds.each do |round|
+      wcif_round = wcif["rounds"][round.number - 1]
+      round.destroy! unless wcif_round
+    end
+
+    # Create missing rounds.
+    wcif["rounds"].each_with_index do |wcif_round, index|
+      round_number = index + 1
+      unless self.rounds.find_by_number(round_number)
+        self.rounds.create!(number: round_number, format_id: wcif_round["format"])
+      end
+    end
+
+    # Update all rounds.
+    wcif["rounds"].each_with_index do |wcif_round, index|
+      round_number = index + 1
+      round = self.rounds.find_by_number(round_number)
+      round.load_wcif!(wcif_round)
+    end
```

Comment:
Ah, this actually doesn't work! Consider the case where you add two rounds of 3x3:

1. 3x3 round 1 with 16 people advancing to round 2
2. 3x3 round 2 (the final round)

If you create them in this order, our round validations will fail while creating 3x3 round 1 (because [final rounds are not allowed to have a "to advance requirement"](https://github.com/thewca/worldcubeassociation.org/blob/54b9f1a42b68b3600338c215c391e17935db522a/WcaOnRails/app/models/round.rb#L25-L29)). The current implementation avoids this problem by first creating all the rounds before setting their attributes.

There is a test for this, I promise =)